spec.md

```diff
@@ -1517,8 +1566,8 @@ for performing primitive operations.
 ## Unsigned Integers
 
 A literal unsigned integer can be created given a non-negative integer value and
-an optional positive bit width. The following example creates a 10-bit unsigned
-integer representing the number 42.
+an optional positive bit width. Integer literals are of constant type.  The 
+following example creates a 10-bit unsigned integer representing the number 42.
 
 ``` firrtl
```

Comment:
Is this equivalent to `const UInt<10>(42)`? Or is it illegal to put the const in front (part of the type only not of the value)?

Reply:
You wouldn't be able to do this. It's saying that `Uint<10>(42)` has type `const Uint<10>`.

Using a Scala-like syntax for type annotations. Before the type was:

```
node a: Uint<1> = Uint<1>(42)
```

Now the type is:

```
node a: const Uint<1> = Uint<1>(42)
```

Reply:
Yes.  I was a bit undecided what to do here.  I decided that the best interpretation is that `UInt<10>(42)` is a primitive operator with the same name as the type it produces, currently, so it could have it's type changed without change in syntax.  Since const decay to non-const, this doesn't change existing usage.